### ISSUE TITLE
Clean up Makefile

### ISSUE THREAD

**Issue by FrankD412** [LLNL/maestrowf#7]
The current default make command installs the full set of development packages. Clean up the Makefile to make it less monolithic in its install process.

**Comment by IanLee1521** [comment]
I would recommend changing the `make wheel` target into `make release` that handles building both sdist and wheel formats:

```
release:
    python setup.py sdist bdist_wheel
```

Additionally, you may not want to tie being in that particular virtual environment into you `Makefile`. Reason being, others (aka I) use things like `virtualenv-wrapper` which would put the virtualenv in a different location, but all of your make targets assume that I'm using the one in `./venv`

Finally, and @cleary3 could comment on this better, I think the better way to say "this target forces you to run this other target" is:

```
cleanall: clean
    rm -rf build/
    rm -rf dist/
    rm -rf $(EGG)
```

**Comment by cleary3** [comment]
> Finally, and @cleary3 could comment on this better, I think the better way to say "this target forces you to run this other target" is:
>
> `cleanall: clean`

Yeah, that's exactly right. Anytime you want a make target to execute another target, you can simply list that subordinate target as a prerequisite after the colon, just as @IanLee1521 indicates.

Additionally, since the `cleanall` and `clean` targets don't actually create any files, you'll want to specify them as `PHONY` targets, thusly:

    .PHONY : 	cleanall \
    		clean

**Comment by FrankD412** [comment]
@IanLee1521, what would a good way to work installation be? I agree that the automatic virtual environment creation is probably not the best way to handle it. I'm airing on the side of assuming that the end user will create their own virtual environment before installing the package.

**Comment by IanLee1521** [comment]
I usually just saw something in the README, e.g. https://github.com/llnl/python-lora#getting-started

**Comment by FrankD412** [comment]
Got it. I'm working on making some tweaks and will push a branch/PR in a few minutes.

**Comment by FrankD412** [comment]
[Branch](https://github.com/LLNL/maestrowf/tree/makefile_cleanup) is located here for this issue.

**Comment by FrankD412** [comment]
I think that the new Makefile in the branch listed above is about as clean as I can get it. Thoughts? @IanLee1521, @cleary3

**Comment by IanLee1521** [comment]
I'll look at it once you start the pull request, where I could comment on it ;)

**Comment by FrankD412** [comment]
Closed by PR #12.
